### PR TITLE
docs: add code style section for javascript

### PR DESCRIPTION
**Pull request by jonoliver**
Adds a code style section for JavaScript! 🎉

### CODE REVIEW

File: code-style/javascript/README.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+-----
+
+Consistent code styling helps our teams work together better and can help prevent common JavaScript pitfalls. We rely heavily on manual code reviews to ensure the quality and functionality of the code we're writing, but prefer to automate code style enforcement as much as possible.
+We use [ESLint][eslint] with a [custom configuration][eslint_config] to help enforce consistent code style.
+
+Here are a few guidelines we like to follow:
+
+- Prefer [ES6][es6] syntax, including
+  - [import][import]
+  - [spread operator][spread]
```
Comment (jonoliver) [inline code-style/javascript/README.md]: Looking for other guidelines to add here

File: code-style/javascript/README.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+We like the dynamic nature of JavaScript, but we also recognize the usefulness of static type checking. We like to use [TypeScript][typescript] when the need arises.
+
+Frameworks / Libraries
+------------------------
+
+JavaScript has a rich ecosystem of libraries. Here are some of the tools we like to use.
+
+- [React][react]
+- [Vue][vue]
+- [Redux][redux]
```
Comment (jonoliver) [inline code-style/javascript/README.md]: What other libraries should be included?

File: code-style/javascript/README.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+Testing
+-------
+
+Automated testing is an important part of our development process. Here are some of the tools we use to test.
```
Comment (jonoliver) [inline code-style/javascript/README.md]: Not sure if I should add some details about testing strategy, or should that be its own section of the standard?

File: code-style/javascript/README.md (hunk 1 above)
Comment (asimpson) [reply code-style/javascript/README.md]: I think I've seen two opinions on spread:
1. it's great, use it everywhere
2. it's great, but quickly becomes hard to follow. Use with caution.

How do we express that? Or am I misremembering?

File: code-style/javascript/README.md (hunk 3 above)
Comment (asimpson) [reply code-style/javascript/README.md]: I think testing gets it's own section

File: code-style/javascript/README.md (hunk 2 above)
Comment (asimpson) [reply code-style/javascript/README.md]: do we want to include things like axios/lodash/moment?

File: code-style/javascript/README.md (hunk 3 above)
Comment (jonoliver) [reply code-style/javascript/README.md]: Cool. I think we can leave this for now and then link to the testing section from here once that's in place.

File: code-style/javascript/README.md (hunk 1 above)
Comment (jonoliver) [reply code-style/javascript/README.md]: This is a good point, I think that the biggest downside of using spread is potential performance implications.

https://medium.com/@eghbali/what-is-the-es6-spread-operator-and-why-you-shouldnt-use-it-57c056078ed9

This touches on a much bigger question: Should we prefer optimizing our code for performance, or readability/maintainability? I tend to prefer maintainability until performance becomes an issue, the exact opposite conclusion the author of the above article comes to 😂. But I don't want my opinion to speak for all of Sparkbox, so I think I'll solicit other opinions in dev.

**Comment by jonoliver** [comment]
@asimpson I pushed a few updates:
- Include additional js libraries
- Add brief descriptions for js libraries
- Remove specific js style guidelines. I think that our eslint config should be the universal source of truth for this.

File: code-style/javascript/README.md [hunk 4]
```diff
@@ -26,23 +26,23 @@ Frameworks / Libraries
...
 
-- [React][react] - A view library for building user interfaces
-- [Vue][vue] - Another view library for building user interfaces
-- [Redux][redux] - A state management library
-- [Axios][axios] - An [isomorphic][isomorphic] HTTP client
-- [Lodash][lodash] - A utility library packed full of useful functionality
-- [Moment][moment] - A date/time library
-- [Chartist][chartist] - A responsive chart library
+- [React][react] – A view library for building user interfaces
+- [Vue][vue] – Another view library for building user interfaces
```
Comment (nrleak) [inline code-style/javascript/README.md]: With the React and Vue descriptions both being essentially the same, would it be helpful to add some information about when we use one vs the other?

File: code-style/javascript/README.md [hunk 5]
```diff
@@ -26,23 +26,23 @@ Frameworks / Libraries
...
 Automated testing is an important part of our development process. Here are some of the tools we use to test.
 
-- [Mocha][mocha] - A test runner
-- [Chai][chai] - A test assertion library
-- [Sinon][sinon] - A test mocking library
-- [Jest][jest] - Another test runner
+- [Mocha][mocha] – A test runner
+- [Chai][chai] – A test assertion library
+- [Sinon][sinon] – A test mocking library
+- [Jest][jest] – Another test runner
```
Comment (nrleak) [inline code-style/javascript/README.md]: Same as the comment above. Would it be helpful to add some more information that differentiates how we use Mocha vs Jest?

Comment (jonoliver) [reply code-style/javascript/README.md]: Updated :)

File: code-style/javascript/README.md (hunk 4 above)
Comment (jonoliver) [reply code-style/javascript/README.md]: I'd like to seek input from some of the devs that are more familiar with Vue and do a follow up PR for this.
